pkg/util/translate/translate.go

```diff
@@ -450,3 +519,25 @@ func KindExists(config *rest.Config, groupVersionKind schema.GroupVersionKind) (
 
 	return metav1.APIResource{}, kerrors.NewNotFound(schema.GroupResource{Group: groupVersionKind.Group}, groupVersionKind.Kind)
 }
+
+func MergeLabelSelectors(elems ...*metav1.LabelSelector) *metav1.LabelSelector {
+	out := &metav1.LabelSelector{}
+	for _, selector := range elems {
+		if selector == nil {
+			continue
+		}
+		for k, v := range selector.MatchLabels {
+			if out.MatchLabels == nil {
+				out.MatchLabels = map[string]string{}
+			}
+			out.MatchLabels[k] = v
+		}
+		for _, expr := range selector.MatchExpressions {
+			if out.MatchExpressions == nil {
+				out.MatchExpressions = []metav1.LabelSelectorRequirement{}
+			}
```

Comment:
is this needed? append should take care of it I think

Reply:
Removed it